pyomo/contrib/incidence_analysis/interface.py

```diff
@@ -101,10 +111,36 @@ def get_bipartite_incidence_graph(variables, constraints, include_fixed=True):
     graph.add_nodes_from(range(M), bipartite=0)
     graph.add_nodes_from(range(M, M + N), bipartite=1)
     var_node_map = ComponentMap((v, M + i) for i, v in enumerate(variables))
-    for i, con in enumerate(constraints):
-        for var in identify_variables(con.expr, include_fixed=include_fixed):
-            if var in var_node_map:
-                graph.add_edge(i, var_node_map[var])
+    vars_in_cons = list(
+        _generate_variables_in_constraints(constraints, include_fixed=True)
+    )
+    to_reset = [
+        var for var in vars_in_cons if var.fixed and var.value is None
+    ]
+    if include_fixed:
+        # AMPLRepnVisitor will not include fixed variables, so if we need to
+        # include these variables, we temporarily unfix them to generate the
+        # AMPLRepn, then re-fix them when we are done.
+        to_unfix = [var for var in variables if var.fixed]
+        context = TemporarySubsystemManager(to_unfix=to_unfix, to_reset=to_reset)
```

Comment:
To preserve the `include_fixed` behavior, we temporarily unfix variables to generate the `AMPLRepn`.